File: 리트코드/0134-gas-station/0134-gas-station.java
class Solution {
    public int canCompleteCircuit(int[] gas, int[] cost) {
        
        // 전체 방문 불가
        if(Arrays.stream(gas).sum() < Arrays.stream(cost).sum())
            return -1;
        
        int start = 0;
        int fuel = 0;
        for(int i = 0; i < gas.length; i++){

            fuel += gas[i] - cost[i];
            if(fuel < 0){
                start = i + 1;
                fuel = 0;
            }

        }
        return start;
    }
}

File: 리트코드/0134-gas-station/0134-gas-station.kt
class Solution {
    fun canCompleteCircuit(gas: IntArray, cost: IntArray): Int {
        if(gas.sum() < cost.sum())
            return -1
        
        var start = 0
        var fuel = 0
        for(i in 0 until gas.size){

            fuel += gas[i] - cost[i]
            if(fuel < 0){
                start = i + 1
                fuel = 0
            }
        }
        return start
    }
}